tests/test_inference.py

```diff
@@ -4453,8 +4453,7 @@ def test_getitem_of_class_raised_type_error(self) -> None:
         # and reraise it as a TypeError in Class.getitem
         node = extract_node(
             """
-        def test():
-            yield
+        def test(): ...
```

Comment:
This changes the function to no longer be a generator right? Why do we want that? https://github.com/pylint-dev/astroid/pull/2424/commits/8a9e944c3acb30592bb8d7754330f107f16eb890 doesn't explain this

Reply:
Good question. Apparently, on 3.13, generators now have `__class_getitem__` (at least according to astroid), defeating the purpose of this test. From the git blame, it's not clear to me why this would need to be a generator in the first place. It seems to just need to be something that raises an `AttributeInferenceError` when you getitem(4) on it.